goonj/channels/slack_channel.py

```diff
@@ -1,22 +1,35 @@
-
 import logging
-import json
+
+import requests
 
 from goonj.channels.base_channel import BaseChannel
+# https://hooks.slack.com/services/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
+from goonj.conf.constants import Sev
+
+logger = logging.getLogger(__name__)
 
 
 class SlackChannel(BaseChannel):
 
-    def send(self, severity, subject, body):
+    def __init__(self, **kwargs):
+        self.name = kwargs['name']
+        self.webhook = kwargs['webhook']
 
-        logger = logging.getLogger(self.__class__.__name__)
+    def send(self, sev, message, subject=None, error_id=None, error=None,
+             tag_list=None):
+
+        if not isinstance(sev, Sev):
+            raise TypeError('Sev must be an instance of type   Sev')
 
         try:
-            final_text = "{sev}: {sub}"
-        #     url = settings.SLACK_CHANNEL_REVIEW_ALERT_URL
-        #     payload = {"text": '`' + str(settings.ENVIRONMENT).upper() + "`: " + final_text}
-        #     json_string = json.dumps(payload, default=lambda o: o.__dict__)
-        #     headers = {'content-type': 'application/json'}
-        #     requests.post(url, data=json_string, headers=headers)
+            headers = {'content-type': 'application/json'}
+            final_text = "Sev: " + sev.value + "\nsubject: " + subject + \
```

Comment:
Use formatting instead of concatenation. Strings are immutable, just eating memory